frontend/src/helpers/LabelFilterHelper.ts

```diff
@@ -83,24 +84,38 @@ const andLabelOperation = (labels: { [key: string]: string }, filters: string[])
   return filterOkForLabel;
 };
 
-const filterLabelByOp = (labels: { [key: string]: string }, filters: string[], op: string = 'or'): boolean => {
+const filterLabelByOp = (labels: { [key: string]: string }, filters: string[], op = 'or'): boolean => {
   return op === 'or' ? orLabelOperation(labels, filters) : andLabelOperation(labels, filters);
 };
 
-export const filterByLabel = (items: itemsType[], filter: string[], op: string = 'or'): itemsType[] => {
+export const filterByLabel = (items: itemsType[], filter: string[], op = 'or'): itemsType[] => {
   return filter.length === 0 ? items : items.filter(item => filterLabelByOp(item.labels, filter, op));
 };
 
-const getKeyAndValues = (filters: string[]): { keys: string[]; keyValues: string[] } => {
+const getKeyAndValues = (filters: string[]): { keyValues: string[]; keys: string[] } => {
   // keys => List of filters with only Label Presence
   // keyValues => List of filters with Label and value
   const keys = filters.filter(f => !f.includes('='));
   const keyValues = filters.filter(f => f.includes('='));
-  return { keys, keyValues };
+  return { keyValues, keys };
+};
+
+export const getGatewayLabels = (labelConfig: string): string[] => {
```

Comment:
Maybe here it can throw some warn/error if the format of the label is not right? If an incorrect value is specified and Kiali is not picking the value because is incorrect, the use is not going to have any warning about it. 

E.g.
![image](https://github.com/kiali/kiali/assets/49480155/ae62d456-2d88-427d-83b0-54eacefcedac)

![image](https://github.com/kiali/kiali/assets/49480155/404e6107-cc07-4458-94c8-626a05abb69c)

But, then, the label is removed: 
![image](https://github.com/kiali/kiali/assets/49480155/b04e951c-69f2-483b-822c-1c424baef447)

Maybe here or while validating the config in https://github.com/kiali/kiali/blob/master/config/config.go#L1187

Reply:
good catch @josunect have added it